File: app/models/qdm/signature.rb
module QDM
  # app/models/qdm/signature.rb
  class Signature < DataElement
    include Mongoid::Document
    embedded_in :patient
    field :type
    field :when
    field :who
    field :onBehalfOf
    field :targetFormat
    field :sigFormat
    field :data
    field :qdmVersion, type: String, default: '4.0.0'
  end
end

File: app/assets/javascripts/Signature.js
const mongoose = require('mongoose');
const { DataElementSchema } = require('./basetypes/DataElement');
const Code = require('./basetypes/Code');
const Interval = require('./basetypes/Interval');
const Quantity = require('./basetypes/Quantity');
const DateTime = require('./basetypes/DateTime');
const { ComponentSchema } = require('./Component');
const { FacilityLocationSchema } = require('./FacilityLocation');
const { IdSchema } = require('./Id');
const Any = require('./basetypes/Any');

const [Number, String] = [
  mongoose.Schema.Types.Number,
  mongoose.Schema.Types.String,
];

const SignatureSchema = DataElementSchema({
  type: Any,
  when: Any,
  who: Any,
  onBehalfOf: Any,
  targetFormat: Any,
  sigFormat: Any,
  data: Any,
  qdmVersion: { type: String, default: '4.0.0' },
  _type: { type: String, default: 'Signature' },

});

module.exports.SignatureSchema = SignatureSchema;
module.exports.Signature = mongoose.model('Signature', SignatureSchema);
